### ISSUE TITLE
Permission problem with docker 'ADD' command

### ISSUE THREAD

**Issue by callmiy** [nickjj/wait-until#1]
I found out that if in my docker file I put:

```
ADD https://raw.githubusercontent.com/nickjj/wait-until/v0.1.0/wait-until /usr/local/bin
RUN chmod +x /usr/local/bin/wait-until
```

I get permission error when I run the `wait-until` script. 

Please note that I am not running as `root` inside the container.

The only way it works is if I give it a `755` permission, and I suggest that the `README` be updated accordingly i.e

```
RUN chmod 755 /usr/local/bin/wait-until
```

**Comment by nickjj** [comment]
Hi,

The `RUN chmod +x /usr/local/bin/wait-until` instruction should make it 755. If you modified your Dockerfile and built the image without that, have you rebuilt it since?

**Comment by callmiy** [comment]
Thanks @nickjj.

A docker file like the below should make the issue clear:

```dockerfile
# run as non root in container, wait-until script will have permission 711
# wait-until script will not work. Error: Permission denied

FROM debian:buster as without-root1

ADD https://raw.githubusercontent.com/nickjj/wait-until/v0.1.0/wait-until /usr/local/bin

RUN groupadd me1 \
    && useradd -m  -g me1 me1 \
    && chmod +x /usr/local/bin/wait-until

USER me1

CMD ["/bin/bash"]

# run as non root in container, give wait-until script explicit 755 permission

FROM debian:buster as without-root2

ADD https://raw.githubusercontent.com/nickjj/wait-until/v0.1.0/wait-until /usr/local/bin

RUN groupadd me2 \
    && useradd -m  -g me2 me2 \
    && chmod 755 /usr/local/bin/wait-until

USER me2

CMD ["/bin/bash"]


# run as root in container, wait-until script works because we run as root

FROM debian:buster as with-root

ADD https://raw.githubusercontent.com/nickjj/wait-until/v0.1.0/wait-until /usr/local/bin

RUN chmod +x /usr/local/bin/wait-until

CMD ["/bin/bash"]
```

Try building each target separately, run and inspect the permissions for the `wait-until` script (to confirm it is `711` in `without-root1` target) and invoke the script. It works in `without-root2` and `with-root` targets, but not in `without-root1`.

**Comment by mzalazar** [comment]
I think it's not a "wait-until" related question.

**Comment by callmiy** [comment]
> I think it's not a "wait-until" related question.

Thanks. But what do you think the problem is?

**Comment by mzalazar** [comment]
Using the first Dockerfile that you provide (FROM debian:buster as without-root1), the script file /usr/local/bin/wait-until has permission 711. When docker downloads the script, it has permission: 700, then you add execute permission (+x) then it would be 711... then you MUST have read permission (r) on that script to execute it (you can do chmod og+rx).
TL;DR You must set READ PERMISSION on the script 😊
Please excuse my english, I'm not good at it.

**Comment by nickjj** [comment]
Thanks. It sounds like this is a documentation problem then.

I should update the docs to explicitly use `755` instead of `+x` for the install notes using `ADD`.

**Comment by callmiy** [comment]
Thanks. @mzalazar  That clears it up.
